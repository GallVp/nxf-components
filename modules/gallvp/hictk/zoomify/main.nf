process HICTK_ZOOMIFY {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/hictk:2.1.4--h061aaa5_0':
        'biocontainers/hictk:2.1.4--h061aaa5_0' }"

    input:
    tuple val(meta), path(hic)

    output:
    tuple val(meta), path("*.hic"), emit: hic
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}.zoomify"
    if( "$hic" == "${prefix}.hic" ) error "Input and output names are the same, use \"task.ext.prefix\" to disambiguate!"
    """
    hictk \\
        zoomify \\
        $args \\
        --threads $task.cpus \\
        --tmpdir ./ \\
        $hic \\
        ${prefix}.hic

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        hictk: \$(hictk --version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}.zoomify"
    if( "$hic" == "${prefix}.hic" ) error "Input and output names are the same, use \"task.ext.prefix\" to disambiguate!"
    """
    echo $args

    touch ${prefix}.hic

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        hictk: \$(hictk --version)
    END_VERSIONS
    """
}
