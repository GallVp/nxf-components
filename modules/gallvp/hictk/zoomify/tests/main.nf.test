
nextflow_process {

    name "Test Process HICTK_ZOOMIFY"
    script "../main.nf"
    config "./nextflow.config"
    process "HICTK_ZOOMIFY"

    tag "modules"
    tag "modules_gallvp"
    tag "hictk"
    tag "hictk/zoomify"
    tag "custom/yahsjuicerpre2pairs"
    tag "yahs/juicerpre"
    tag "hictk/load"

    setup {
        run('YAHS_JUICERPRE') {
            script '../../../yahs/juicerpre/main.nf'
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.bam' , checkIfExists: true)
                ]

                input[1] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.agp' , checkIfExists: true)
                input[2] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.fasta.fai' , checkIfExists: true)
                """
            }
        }

        run('CUSTOM_YAHSJUICERPRE2PAIRS') {
            script '../../../custom/yahsjuicerpre2pairs/main.nf'
            process {
                """
                input[0] = YAHS_JUICERPRE.out.txt

                def sizes = new File('hongyang.sizes')
                sizes.text = "HYh1_Chr01:6000000-6015000\\t15001\\nHYh2_Chr01:6000000-6015000\\t15001\\n"

                input[1] = sizes.toPath()
                """
            }
        }

        run('HICTK_LOAD') {
            script '../../../hictk/load/main.nf'

            process {
                """
                input[0] = CUSTOM_YAHSJUICERPRE2PAIRS.out.pairs
                input[1] = '4dn' // format
                """
            }
        }
    }

    test("hongyang - hic") {

        when {
            process {
                """
                input[0] = HICTK_LOAD.out.hic
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("hongyang - hic - stub") {
        options '-stub'

        when {
            process {
                """
                input[0] = HICTK_LOAD.out.hic
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}