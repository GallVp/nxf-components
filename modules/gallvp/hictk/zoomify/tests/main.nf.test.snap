{
    "hongyang - hic - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.zoomify.hic:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,47dd723e575db6aecd74c7b490cc4ec7"
                ],
                "hic": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.zoomify.hic:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,47dd723e575db6aecd74c7b490cc4ec7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-11T19:54:45.997925"
    },
    "hongyang - hic": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.zoomify.hic:md5,fdf3100373d6683fcdd2ea95a9ba18cc"
                    ]
                ],
                "1": [
                    "versions.yml:md5,47dd723e575db6aecd74c7b490cc4ec7"
                ],
                "hic": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.zoomify.hic:md5,fdf3100373d6683fcdd2ea95a9ba18cc"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,47dd723e575db6aecd74c7b490cc4ec7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-11T19:51:37.966564"
    }
}